sdk/core/azure-core/src/http/curl/curl_connection_pool_private.hpp

```diff
@@ -89,15 +97,17 @@ namespace Azure { namespace Core { namespace Http {
     // Class can't have instances.
     CurlConnectionPool() = delete;
 
+    static void StopCleaner() { g_isCleanConnectionsRunning = false; }
```

Comment:
This doesn't wait for the cleaner thread to terminate, so I believe does not resolve #1717.

Reply:
fixed: https://github.com/Azure/azure-sdk-for-cpp/pull/2105